File: Client/app/shared/patientDiagnosis.ts
import { MedicalInstitution } from './medicalInstitution';
export class PatientDiagnosis{
    Id:string;
    PatientId: string;
    DoctorId: string;
    DoctorName: string;
    MedicalInstitution: MedicalInstitution;
    InOutPatient: boolean;
    DiagnosisDate: Date;
    DischargeDate: Date;
    InclusionDate: Date;
    General: string;
    Symptoms: {Key:string,Symptom:SymptomInfo}[];

    constructor(patientId:string){
        this.PatientId = patientId;
    }
    fromJSON(json:Object) {
        for (var propName in json)
            this[propName] = json[propName];
        return this;
    }
}

export class SymptomInfo{
    SymptomName:string;
    Is:boolean;
    ExtraInfo:string;
    Value:number;

    fromJSON(json:Object) {
        for (var propName in json)
            this[propName] = json[propName];
        return this;
    }
}

File: Server/MRP.DAL/Models/PatientDiagnosis.cs
﻿using MRP.Common.DTO;
using System;
using System.Collections.Generic;

namespace MRP.DAL.Models
{
    public class PatientDiagnosis
    {
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public string DoctorName { get; set; }
        public MedicalInstitution MedicalInstitution { get; set; }
        public bool InOutPatient { get; set; }
        public DateTime DiagnosisDate { get; set; }
        public DateTime DischargeDate { get; set; }
        public DateTime InclusionDate { get; set; }
        public string General { get; set; }
        public Dictionary<string,SymptomInfo> Symptoms { get; set; }
    }
}